### PR TITLE
Allow Rails 7.2

### PR DESCRIPTION
**Pull request by tvdeyen**
## Summary

Allows to update host apps to Rails 7.2.

The only change is that we need to raise the minimum version of Ruby to 3.1, but since it is EOL anyway
it should not be a big deal for anyone.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

**Comment by tvdeyen** [comment]
> How should we handle the error we get with Ransack 4.2 and above? It doesn't look like there has been any movement [on the PR to fix the issue.](https://github.com/activerecord-hackery/ransack/pull/1468)

That's a very good question. I need to read on the issue first in order to make an opinion. Since I use Ransack 4.2 with great success in several other apps and engines I am not sure why this is an issue in Solidus at all.

**Comment by tvdeyen** [comment]
I looked into the problem and I am sure I can replace the custom ransacker it with normal ransack. PR upcoming maybe tomorrow

**Review by MadelineCollier — COMMENTED** [review]
Nice, thanks for addressing the root cause ❤️

**Comment by tvdeyen** [comment]
> Nice, thanks for addressing the root cause ❤️

Should be fixed by https://github.com/solidusio/solidus/pull/5853

**Review by forkata — COMMENTED** [review]
Great work here @tvdeyen! We may be blocked by the `solidus_starter_frontend` that is causing issues bundling due to `cannonical_rails` 😞 I believe there hasn't been a release that includes [this PR](https://github.com/jumph4x/canonical-rails/pull/92) yet. There is an open issue for it [here](https://github.com/jumph4x/canonical-rails/pull/93).

File: .circleci/config.yml [hunk 1]
```diff
@@ -191,10 +191,10 @@ commands:
     parameters:
       app_root:
         type: string
-        default: '/tmp/my_app'
+        default: "/tmp/my_app"
```
[the remark this answers is not in the document]
Comment (tvdeyen) [reply .circleci/config.yml:194]: Good call!

**Comment by tvdeyen** [comment]
> @tvdeyen I completely agree with you on the unnecessary dependency on `canonical_rails`. I would be happy to support with trying to remove that dependency from the starter frontend if you don't have time to tackle that?

@forkata sure, that would be really helpful!

**Comment by jumph4x** [comment]
Apologies for the delays on this. Getting it handled right now.

**Comment by tvdeyen** [comment]
> Apologies for the delays on this. Getting it handled right now.
> 
> Though I understand it might get dropped as a dep. No matter, releasing a patch bump is a good idea: https://rubygems.org/gems/canonical-rails/versions/0.2.16

Thanks,  Denis! No worries.

**Comment by tvdeyen** [comment]
@solidusio/core-team this is now ready for review. 

We still need a merge ofhttps://github.com/solidusio/solidus_starter_frontend/pull/374 that allows Rails 7.2 in order to make the solidus-installer tests pass in this branch. This is kind of a chicken egg situation. The PR in solidus_starter_frontend runs with this branch to make it's tests green and vice versa. I still think we should merge the solidus_starter_frontend  PR first so [we can revert back to main in the starter frontend app template](https://github.com/tvdeyen/solidus/blob/rails-7.2/core/lib/generators/solidus/install/app_templates/frontend/starter.rb) and [this change in the solidus-starter PR](https://github.com/solidusio/solidus_starter_frontend/pull/374/files#diff-f365495e9d50595dce2a70fd2b44722f837b710dafda7d0eafed7315c2e71110R55-R56)

WDYT?

**Comment by tvdeyen** [comment]
I extracted the Ruby 3.1 bump [into a separate PR ](https://github.com/solidusio/solidus/pull/5891)for better visibility

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tvdeyen) [reply .circleci/config.yml:195]: Done in a9c88c9f0a46d27de3e42cf9bcaf720eecbc9bad

**Review by jarednorman — APPROVED** [review]

**Pull request merged**